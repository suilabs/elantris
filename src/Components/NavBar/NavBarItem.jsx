import React from 'react';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './NavBarItem.scss';

const NavBarItem = (props) => {
  const root = props.history.location.pathname.split('/')[1];
  const active = `/${root}` === props.to ? '--active' : '';
  const classname = `sui-navbar__item${active} ${props.className}`;
  return (
    <li className={classname}>
      <Link id={props.to} to={props.to} className="sui-navbar__item--link">
        {props.label}
      </Link>
    </li>
  );
};

NavBarItem.propTypes = {
  to: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  className: PropTypes.string,
};

NavBarItem.defaultProps = {
  className: 'menu-item',
};

export default withRouter(NavBarItem);
