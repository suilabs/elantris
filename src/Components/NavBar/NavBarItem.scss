@import '../../base.scss';

.sui-navbar__item {
  display: inline;

  a {
    transition: border-bottom $sui-duration-medium ease;
    border-bottom: pixels(2) solid $sui-color-primary-total-trans;
  }

  &--active {
    display: inline;

    a {
      transition: border-bottom $sui-duration-medium ease;
      border-bottom: pixels(2) solid $sui-color-primary;
    }
  }

  &--link {
    @include link;
  }
}
