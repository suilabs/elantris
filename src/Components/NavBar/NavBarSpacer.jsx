import React from 'react';

import './NavBarSpacer.scss';

export default () => (
  <li className="sui-navbar__item-spacer">
    Â·
  </li>
);
