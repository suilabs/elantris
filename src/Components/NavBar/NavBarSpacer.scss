.sui-navbar__item-spacer {
  display: inline;
  padding: 0 .25rem;
}

