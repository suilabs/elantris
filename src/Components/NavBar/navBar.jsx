import React from 'react';
import PropTypes from 'prop-types';

import './navBar.scss';

const NavBar = ({ children = [] }) => (
  <nav className="sui-navbar">
    <ul>
      {children}
    </ul>
  </nav>
);

NavBar.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node]).isRequired,
};


export default NavBar;
