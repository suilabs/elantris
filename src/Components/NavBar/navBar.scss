@import '../../base.scss';

.sui-navbar {
  left: 50%;
  display: inline-block;
  padding-top: 1rem;
  text-transform: capitalize;

  @include break-on(mobile) {
    left: auto;
    width: 100%;
  }
}
