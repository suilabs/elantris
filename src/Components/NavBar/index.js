import NavBar from './navBar';
import NavBarItem_ from './NavBarItem';
import NavBarSpacer_ from './NavBarSpacer';

export const NavBarItem = NavBarItem_;
export const NavBarSpacer = NavBarSpacer_;
export default NavBar;
