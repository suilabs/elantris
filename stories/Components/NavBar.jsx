import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';

import NavBar, { NavBarItem } from '../../src/Components/NavBar';

storiesOf('NavBar', module)
  .add('Normal', () => {
    window.suilabs = { isMobile: false };
    return (
      <BrowserRouter>
        <NavBar>
          <NavBarItem to="#" label="Design" />
          <NavBarItem to="#" label="Software" />
          <NavBarItem to="#" label="About us" />
        </NavBar>
      </BrowserRouter>
    );
  });
