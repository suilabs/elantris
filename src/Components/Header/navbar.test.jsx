import React from 'react';

import NavBar from './navBar';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <NavBar/>
    );
    
    expect(tree).toMatchSnapshot();
  });
});
