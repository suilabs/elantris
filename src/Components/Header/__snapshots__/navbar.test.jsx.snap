// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "toJson": [Function],
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <withRouter(NavBarItem)
                label="Design"
                to="/design"
        />
        <Unknown />
        <withRouter(NavBarItem)
                label="Software"
                to="/software"
        />
        <Unknown />
        <withRouter(NavBarItem)
                label="About us"
                to="/about"
        />
        <Unknown />
</ul>,
      "className": "sui-navbar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(NavBarItem)
            label="Design"
            to="/design"
/>,
          <Unknown />,
          <withRouter(NavBarItem)
            label="Software"
            to="/software"
/>,
          <Unknown />,
          <withRouter(NavBarItem)
            label="About us"
            to="/about"
/>,
          <Unknown />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Design",
            "to": "/design",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Software",
            "to": "/software",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "About us",
            "to": "/about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <withRouter(NavBarItem)
                    label="Design"
                    to="/design"
          />
          <Unknown />
          <withRouter(NavBarItem)
                    label="Software"
                    to="/software"
          />
          <Unknown />
          <withRouter(NavBarItem)
                    label="About us"
                    to="/about"
          />
          <Unknown />
</ul>,
        "className": "sui-navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(NavBarItem)
              label="Design"
              to="/design"
/>,
            <Unknown />,
            <withRouter(NavBarItem)
              label="Software"
              to="/software"
/>,
            <Unknown />,
            <withRouter(NavBarItem)
              label="About us"
              to="/about"
/>,
            <Unknown />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "Design",
              "to": "/design",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "Software",
              "to": "/software",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "About us",
              "to": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
